console/console-init/ui/src/modules/iot-device/components/DeviceFilter/DeviceFilter.tsx

```diff
@@ -220,6 +288,80 @@ const DeviceFilter: React.FunctionComponent<IDeviceFilterProps> = ({
         <FormGroup label="Added date" fieldId="filter-device-added-date">
           <DateFilterSection filter={filter} setFilter={setFilter} />
         </FormGroup>
+        <FormGroup
+          label="Gateway group membership"
+          fieldId="device-filter-gateway-membership-group-input"
+        >
+          <GatewayGroupTypeAheadSelect
+            id="add-gateway-group-membership-typeahead-select"
+            aria-label="gateway group membership dropdown"
+            aria-describedby="multi typeahead for gateway groups membership"
+            onSelect={onSelectGatewayGroup}
+            onClear={onClear}
+            selected={gatewayGroups}
+            typeAheadAriaLabel={"typeahead to select gateway group membership"}
+            isMultiple={true}
+            placeholderText={"Input gateway group name"}
+          />
+        </FormGroup>
+
+        <FormGroup
+          label="Gateway connections"
+          fieldId="device-gateway-connection-dropdowntoggle"
+        >
+          <Grid>
+            <GridItem span={4}>
+              <DropdownWithToggle
+                id={"device-filter-gateway-conn-dropdown"}
+                toggleId="device-filter-gateway-conn-dropdowntoggle"
+                name="Device Connections"
+                className={css(styles.dropdown_align)}
+                toggleClass={css(styles.dropdown_toggle_align)}
```

Comment:
Instead of having the entire device Filter UI on the same component, please create smaller and simpler components and move its handlers on those components. And import those components in Device Filter.